cachi2/core/errors.py

```diff
@@ -1,107 +1,100 @@
 import textwrap
 from typing import ClassVar, Optional
 
+_argument_not_specified = "__argument_not_specified__"
+
 
 class Cachi2Error(Exception):
     """Root of the error hierarchy. Don't raise this directly, use more specific error types."""
 
     is_invalid_usage: ClassVar[bool] = False
+    default_solution: ClassVar[Optional[str]] = None
+
+    def __init__(
+        self,
+        reason: str,
+        *,
+        solution: Optional[str] = _argument_not_specified,
+        docs: Optional[str] = None,
+    ) -> None:
+        """Initialize a Cachi2 error.
+
+        :param reason: explain what went wrong
+        :param solution: politely suggest a potential solution to the user
+        :param docs: include a link to relevant documentation (if there is any)
+        """
+        super().__init__(reason)
+        if solution == _argument_not_specified:
+            self.solution = self.default_solution
+        else:
+            self.solution = solution
+        self.docs = docs
 
     def friendly_msg(self) -> str:
         """Return the user-friendly representation of this error."""
-        return str(self)
+        msg = str(self)
+        if solution := self.solution:
```

Comment:
`if self.solution:` would be probably sufficient here. :)

Reply:
True, I was just too lazy to type `self.` twice when moving the code out of the function :smile: